compiler/AST/iterator.cpp

```diff
@@ -645,8 +696,9 @@ buildHasMore(IteratorInfo* ii, BlockStmt* singleLoop) {
     // In copied expressions, replace _ic with hasMore->_this .
     map.put(ic, ii->hasMore->_this);
 
-    // NOAKES 2014/12/04 Why is this copy here?
-    testBlock = cforLoop->testBlockGet()->copy(&map);
+    // 2015-02-23 hilde: There was an unnecessary call to copy(&map) here --
+    // pointed out by mnoakes -- which I removed.
+    testBlock = cforLoop->testBlockGet();
```

Comment:
I'm not sure the comment about what used to be there is needed -- I don't feel strongly about that though

pinging @noakesmichael that the copy is being removed since his comment was attached to it